cloud/linode/loadbalancers.go

```diff
@@ -67,6 +70,62 @@ func newLoadbalancers(client *linodego.Client, zone string) cloudprovider.LoadBa
 	return &loadbalancers{client: client, zone: zone}
 }
 
+func (l *loadbalancers) getNodeBalancerForService(ctx context.Context, service *v1.Service) (*linodego.NodeBalancer, error) {
+	rawID, _ := getServiceAnnotation(service, annLinodeNodeBalancerID)
+	id, idErr := strconv.Atoi(rawID)
+	hasIDAnn := idErr == nil && id != 0
+
+	if hasIDAnn {
```

Comment:
The reason why I didn't exit early is so that if the NodeBalancer specified by the ID does not exist, it falls back to looking for the NodeBalancer by the most recent status.